controllers/cluster_controller.go

```diff
@@ -262,9 +270,135 @@ func (r *ClusterReconciler) postClusterProviderReconcile(ctx context.Context, lo
 		}
 	}
 
+	if cluster.IsPackagesEnabled() {
+		if err := r.postReconcilePackagesForWorkloadCluster(ctx, log, cluster); err != nil {
+			return controller.Result{}, err
+		}
+	}
+
 	return controller.Result{}, nil
 }
 
+// PackagesClient handles curated packages operations from within the cluster
+// controller.
+type PackagesClient interface {
+	EnableFullLifecycle(context.Context, logr.Logger, string, string, *v1alpha1.Image, *registrymirror.RegistryMirror, ...curatedpackages.PackageControllerClientOpt) error
+	ReconcileDelete(context.Context, logr.Logger, client.Client, *anywherev1.Cluster) error
+}
+
+func (r *ClusterReconciler) postReconcilePackagesForWorkloadCluster(ctx context.Context, log logr.Logger, cluster *anywherev1.Cluster) (err error) {
```

Comment:
Could we have a guard at the beggining of this method that returns early if "full lifecycle" is already enabled?

That allows to avoid writing the kubeconfig to disk and all that jazz if we already "enabled" curated packages for that cluster.

Reply:
So what I mean to say by the above, is that I think your concern about writing kubeconfig when it isn't necessary is moot now, since that logic is removed. If you have another issue, or I've misunderstood, please let me know.